testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +126,35 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_nodensity(self, D):
+        del D.units['density']
+        units = {'length': 'A'}
+        D._check_set_unit(units)
+
+    def test_convert_length(self, D):
+        D.convert_length()
+
+    def test_return_density_and_length_units(self, D):
+        units = {'density': 'A^{-3}', 'length': 'A'}
+        D.convert_length(units['length'])
+        D.convert_density(units['density'])
+
+    def test_ifnot_density(self, D1):
+        with pytest.raises(RuntimeError, match='The grid is not'):
+            D1.convert_density()
+
+    def test_convert_density_valueError(self, D):
+        with pytest.raises(ValueError, match='The name of the unit'):
+            D.convert_density()
+
+    def test_convert_density_units(self, D):
+        unit = 'nm^{-3}'
+        D.convert_density(unit)
+
+    def test_repr_(self, D, hist):
+        D.__repr__()
+        hist.__repr__()
```

Comment:
Similar to Irfan's other comments--this just produces two strings so doesn't do much apart from flushing code paths. Making some kind of assertion about the contents of the strings seems like it might be in order for this case?

Reply:
Thanks, you're correct. I rewrote all the test for `__repr__`.